heareval/tasks/pipeline.py

```diff
@@ -291,6 +293,11 @@ def split_train_test_val(self, metadata: pd.DataFrame):
 
     def run(self):
         process_metadata = self.get_process_metadata()
```

Comment:
There are a handful of things I hate about this :(

ExtractMetadata.get_process_metadata() is a generic method but:
* there is no comment describing WHAT this method should do, and when it should be overriden
* the generic method doesn't use slugify_file_name. Why?

This seems confusing and risky and bug prone

Reply:
I agree -- this needs to be documented better - and probably simplified. But let's save the refactor for a different PR.

Reply:
```
    def get_process_metadata(self) -> pd.DataFrame:
        """
        Return a dataframe containing the task metadata for this
        entire task.
        By default, we do one split at a time and then concat them.
        This runs only for the splits which have a requires task. All
        other splits are sampled with the split train test val function.
        You might consider overriding this for some datasets (like
        Google Speech Commands) where you cannot process metadata
        on a per-split basis.
        """
```
The above description gives the information about this method, and also when to override it. But we can be more specific.

Reply:
@jorshi and @khumairraj yes okay let's postpone refactor: https://github.com/neuralaudio/hear-eval-kit/issues/208